scripts/weekly-dep-report.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# this script is making an assumption that the repository contains at least two commits
+
+main_branch=${1:-"HEAD"}
+git checkout --quiet "${main_branch}"
+
+last_commit=$(git log --pretty=format:'%H' | head -n 1)
+first_commit_within_last_seven_days=$(git log --since=7.days --reverse --pretty=format:"%H" | head -n 1)
+
+format_change_in_dependency_output () {
+    #if negative, do nothing; if positive, add + prefix; if zero, return "="
+    if [[ $1 > 0 ]]; then
+        echo "+$1"
+    elif [[ $1 == 0 ]]; then
+        echo "="
+    else
+        echo "$1"
+    fi
+}
+
+git checkout --quiet "${last_commit}"
+total=$(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind thirdparty --omit-edges-into diem-workspace-hack --omit-edges-into libra-workspace-hack| wc -l | tr -d ' ')
+direct=$(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind directthirdparty --omit-edges-into diem-workspace-hack --omit-edges-into libra-workspace-hack | wc -l | tr -d ' ')
+lsr_total=$(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind thirdparty --omit-edges-into diem-workspace-hack --omit-edges-into libra-workspace-hack -p safety-rules | wc -l | tr -d ' ')
+lsr_direct=$(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind directthirdparty --omit-edges-into diem-workspace-hack --omit-edges-into libra-workspace-hack -p safety-rules | wc -l | tr -d ' ')
+dups=$(cargo +stable guppy dups --target x86_64-unknown-linux-gnu | wc -l | tr -d ' ')
```

Comment:
Similar to how you made a function for the format_change_in_dependency_output, can you make functions for each of these so that we don't have to duplicate this below?